python_modules/dagster/dagster/_core/definitions/auto_materialize_rule.py

```diff
@@ -215,6 +257,53 @@ def description(self) -> str:
         """
         ...
 
+    def merge_with_past_results(
+        self,
+        context: RuleEvaluationContext,
+        asset_partitions_by_evaluation_data: Mapping[
+            Optional[AutoMaterializeRuleEvaluationData], AbstractSet[AssetKeyPartitionKey]
+        ],
+        evaluated_candidates: Optional[AbstractSet[AssetKeyPartitionKey]] = None,
+    ) -> RuleEvaluationResults:
+        """Merge the new results of an evaluation with any previous results which are still valid,
```

Comment:
What exactly does "merge" mean? And what about "still valid" mean? I'm guessing that "merge" means taking the union of the asset partitions?

Reply:
maybe:

```
Combine the new results of an evaluation with any results from the previous tick which are still relevant.
```

or

```
Combine the new results of an evaluation with any results from the previous tick for asset partitions which are still unhandled.
```